src/fdc/DiskManipulator.cc

```diff
@@ -186,18 +186,20 @@ void DiskManipulator::execute(std::span<const TclObject> tokens, TclObject& resu
 		create(tokens);
 
 	} else if (subCmd == "format") {
-		bool dos1 = false;
 		string_view drive = tokens[2].getString();
+		MSXBootSectorType bootType = MSXBootSectorType::DOS2;
 		if (tokens.size() == 4) {
-			if (drive == "-dos1") {
-				dos1 = true;
+			if (drive == one_of("-dos1", "-dos2", "-extended")) {
+				bootType = drive == "-dos1" ? MSXBootSectorType::DOS1 :
+					drive == "-dos2" ? MSXBootSectorType::DOS2 : MSXBootSectorType::EXTENDED;
 				drive = tokens[3].getString();
-			} else if (tokens[3] == "-dos1") {
-				dos1 = true;
+			} else if (tokens[3] == one_of("-dos1", "-dos2", "-extended")) {
+				bootType = tokens[3] == "-dos1" ? MSXBootSectorType::DOS1 :
+					tokens[3] == "-dos2" ? MSXBootSectorType::DOS2 : MSXBootSectorType::EXTENDED;
 			}
```

Comment:
This was already a problem in the code before your changes. There is no good error checking, for example an invalid flag in the last position is interpreted as `DOS2` instead of reported as an error.

Here's one way to improve this (not tested):
```
if (tokens.size() == 4) {
    auto parseType = [](std::string_view s) -> std::optional<MSXBootSectorType> {
        if (s == "-dos1")     return MSXBootSectorType::DOS1;
        if (s == "-dos2")     return MSXBootSectorType::DOS2;
        if (s == "-extended") return MSXBootSectorType::EXTENDED;
        return {};
    };
    if (auto t = parseType(drive)) {
        bootType = *t;
        drive = tokens[3].getString();
    } else {
        auto t = parseType(tokens[3].getString())) {
        if (!t) {
            throw ... some error message ...
        }
        bootType = *t;
    }
}
```
This also has the advantage that we don't repeat the flag names 4(!) times. (And reduce the number of string comparisons).

If this gets any more complicated we might want to use the `parseTclArgs()` utility function (though at the moment it doesn't have good support yet for enum-like flags).

Could you also update the help text around line 255. And the manual in `doc/manual/diskmanipulator.html`.